api/handlers/dataexplorerhandler.py

```diff
@@ -305,7 +305,39 @@ def _parse_request(self, request_type='search'):
         for f in filters:
             if f.get('terms'):
                 for k,v in f['terms'].iteritems():
-                    modified_filters.append({'terms': {k+'.raw': v}})
+                    if "null" in v:
+                        try:
+                            v.remove("null")
+                        except AttributeError:
+                            v = None
+                        null_filter = {
+                            'bool': {
+                                'should': [
+                                    {
+                                        'bool': {
+                                            'must': [
+                                                {
+                                                    'bool': {
+                                                        'must_not': [
+                                                            {
+                                                                'exists': {'field': k}
+                                                            }
+                                                        ]
+                                                    }
+                                                },
+                                                {"exists":{"field":k.split('.')[0]}}
```

Comment:
It should be rare that they would reference a top-level key like `container_type`, but we might want to be a bit more fault tolerant if they do reference a top level key. In that case, this line is unnecessary.